Update glm robust with outliers and add extra dependency template

OriolAbril opened the pull request:
I started with a couple build warnings, then moved to fixing a couple lines in the notebook as it was
generating warnings due to consecutive horizontal lines and other formatting nits, and
ended up rerunning the notebook, updating all plotting and generating a template proposal for
extra dependencies in notebooks. The dependencies template is related to #200 

+ [x] Notebook follows style guide https://github.com/pymc-devs/pymc/wiki/PyMC-Jupyter-Notebook-Style-Guide
+ [x] PR description contains a link to the relevant issue: https://github.com/pymc-devs/pymc-examples/issues/87

OriolAbril commented:
Here is the [preview](https://pymc--249.org.readthedocs.build/projects/examples/en/249/generalized_linear_models/GLM-robust-with-outlier-detection.html) (@jonsedar might be interested? I realized there were some mistakes with ArviZ usage in the latest update, that I actually reviewed but not too well, everything is fixed now I believe), and the specific link to the [extra installs section](https://pymc--249.org.readthedocs.build/projects/examples/en/249/generalized_linear_models/GLM-robust-with-outlier-detection.html#installation-notes) (the attention and dropdown are the template. 

I have added:

```
:::{include} extra_installs.md
:::
```

in a markdown cell plus a notebook metadata key that says "extra_dependencies: seaborn" and all the commands and text are imported from the template. cc @martinacantaro @chiral-carbon @Sayam753

jonsedar commented:
Can it really be six years? Looks pretty good, I really like the book styling - what is that?

I've an action unfortunately very far down my todo list to create more...

OriolAbril commented:
> I really like the book styling - what is that?

What part of the styling do you mean? I think the only common point between how notebooks were built around the beginning of this year and now is that we still use sphinx. 

* We changed the theme. It is no longer a pymc specific theme but the https://pydata-sphinx-theme.readthedocs.io/en/latest/, used also in several other projects. This ensures the theme will continue to look good and work without us intervening, unlike before where issues with the theme like the search bar not working were on us and took a long time to fix if they were fixed at all.
* We changed the notebook rendering engine from nbsphinx to [myst-nb](https://myst-nb.readthedocs.io/en/latest/). This allows us to take advantage of all sphinx features and extensions also within the notebooks. We can then for example use sphinxcontrib-bibtex to cite like in latex and generate auto intext citations with customizable format and automatic bibliography sections. 
* We have integrated several extra extensions with the build. Most notably [ablog](https://ablog.readthedocs.io/en/latest/) which handles tags and categories with notebooks to improve navigation and notebook finding. But also sphinx-design for cool formatting options like dropdows or tabs to compare alternatives, bibtex support as mentioned, or sphinx-codeautolink in order to autoupdate some objects in the code blocks to hyperlinks to their documentation.
* We also added extra custom templates to extend the theme like the recommended citation at the bottom (mostly automatic) or the github/binder/colab badges (completely automatic).

martinacantaro commented on CONTRIBUTING.md:
````diff
@@ -27,10 +27,13 @@ Therefore, all notebooks will be progressively updated along this path:
 ```
 
+See https://github.com/pymc-devs/pymc-examples/wiki/Notebook-updates-overview for a more detailed
+description of what each of the status mean.
````
what each of the statuses means

martinacantaro commented on examples/generalized_linear_models/GLM-robust.ipynb:
```diff
@@ -482,7 +482,7 @@
    "nbconvert_exporter": "python",
    "pygments_lexer": "ipython3",
-   "version": "3.8.10"
+   "version": "3.9.7"
```
When I look at the RTD preview from the PR, I still see 3.8 in the Watermark section. Is that OK? Will it be updated automatically when merging?

OriolAbril replied:
I should remove this file from the PR, I opened it by mistake and must have saved when closing out of muscle memory, it has not been edited nor executed and the watermark is right. The metadata version is not a trustworthy source of information on what was used to run the code.
